### PR TITLE
Use already loaded xml resources

### PR DESCRIPTION
**Pull request by lexs**
Previously xml files we had already loaded would be reparsed instead of being fetched from the ResourceLoader.

This commit fixes this and:
* Consolidates ResBunch and ResBundle
* Refactors xml handling to XmlBlock

After this commit I want to investigate simply caching by file name instead, this would enable us to lazy load resources instead.

### CODE REVIEW

File: robolectric-resources/src/main/java/org/robolectric/res/RoutingResourceLoader.java [hunk 1]
```diff
@@ -37,8 +36,8 @@ public Plural getPlural(ResName resName, int quantity, String qualifiers) {
   }
 
   @Override
-  public Document getXml(ResName resName, String qualifiers) {
-    return pickFor(resName).getXml(resName, "");
+  public org.robolectric.res.builder.XmlBlock getXml(ResName resName, String qualifiers) {
```
Comment (jongerrish) [inline robolectric-resources/src/main/java/org/robolectric/res/RoutingResourceLoader.java]: Does it need to be fuilly qualified?

File: robolectric-resources/src/main/java/org/robolectric/res/builder/XmlBlock.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package org.robolectric.res.builder;
+
+import org.robolectric.res.Attribute;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class XmlBlock {
```
Comment (jongerrish) [inline robolectric-resources/src/main/java/org/robolectric/res/builder/XmlBlock.java]: Can you add some Javadoc, not sure what an XmlBlock is

File: robolectric-resources/src/main/java/org/robolectric/res/RoutingResourceLoader.java (hunk 1 above)
Comment (lexs) [inline robolectric-resources/src/main/java/org/robolectric/res/RoutingResourceLoader.java]: No, IntelliJ refactor weirdness.

File: robolectric-resources/src/main/java/org/robolectric/res/builder/XmlBlock.java (hunk 2 above)
Comment (lexs) [inline robolectric-resources/src/main/java/org/robolectric/res/builder/XmlBlock.java]: Done.

File: robolectric-resources/src/main/java/org/robolectric/res/ResourceLoader.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
   Plural getPlural(ResName resName, int quantity, String qualifiers);
 
-  Document getXml(ResName resName, String qualifiers);
+  org.robolectric.res.builder.XmlBlock getXml(ResName resName, String qualifiers);
```
Comment (ChristianBecker) [inline robolectric-resources/src/main/java/org/robolectric/res/ResourceLoader.java]: Again some refactor-weirdness?

Comment (lexs) [inline robolectric-resources/src/main/java/org/robolectric/res/ResourceLoader.java]: Yes, made a search now to actually make sure there wasn't any more.

**Comment by erd** [comment]
Could you rebase this?

**Pull request merged**